.gitlab-ci.yml

```diff
@@ -1,41 +1,29 @@
-image: python:3.7
-
-stages:
-  - deploy
-  - deploy-latest
-
-deploy:
-  stage: deploy
+# This file is a template, and might need editing before it works on your project.
+build-master:
+  # Official docker image.
+  image: docker:latest
+  stage: build
+  services:
+    - docker:dind
```

Comment:
we don't need `dind`, if we can build the image using `kaniko`

I don't see the reason to go a step back here.

Reply:
But you are adding another layer of complexity without using any of the benefits of kaniko or without any problem which needs to be solved by using kaniko.

So maybe i was just confused, and missed the "latest" image in your build process and image registry, and this "deploy" was also irritating, because ... i don't know, i missed something, i expected this pipeline just builds the image for every tag and branch and pushes it into the registry. But if there is more, than, well.

Reply:
The latest image will be added automatically as soon as a new tag is created.

I don't think we need an image for every branch.

Reply:
See: https://github.com/python-gitlab/python-gitlab/commit/01cbc7ad04a875bea93a08c0ce563ab5b4fe896b